src/os/bluestore/BlueFS.cc

```diff
@@ -731,25 +746,45 @@ void BlueFS::_init_alloc()
   if (bdev[BDEV_SLOW]) {
     alloc_size[BDEV_DB] = cct->_conf->bluefs_alloc_size;
     alloc_size[BDEV_SLOW] = shared_alloc_size;
+    change_alloc_size(super.bluefs_max_alloc_size[BDEV_DB],
+                      cct->_conf->bluefs_alloc_size, alloc_size_changed);
+    change_alloc_size(super.bluefs_max_alloc_size[BDEV_SLOW],
+                      shared_alloc_size, alloc_size_changed);
   } else {
     alloc_size[BDEV_DB] = shared_alloc_size;
     alloc_size[BDEV_SLOW] = 0;
+    change_alloc_size(super.bluefs_max_alloc_size[BDEV_DB],
+                      shared_alloc_size, alloc_size_changed);
+  }
+  if (alloc_size_changed) {
```

Comment:
What is it here for? Why don't you use the second call of _write_super() below only?

Reply:
_write_super() has the param named dev, it indicates which bdev to store super. I think there are two ways, calling _write_super() twice or adding another variable to indicate BDEV_DB or BDEV_NEWDB. Am i correct, which way is better?

Reply:
> What is it here for? Why don't you use the second call of _write_super() below only?

_write_super() has the param named dev, it indicates which bdev to store super. I think there are two ways, calling _write_super() twice or adding another variable to indicate BDEV_DB or BDEV_NEWDB. Am i correct, which way is better?

Reply:
ah... sorry, I missed that these calls get different parameters. Please ignore.